packages/astro/src/assets/fonts/utils.ts

```diff
@@ -264,3 +277,83 @@ export function kebab(value: string) {
 export function getFamilyName(family: Pick<FontFamilyAttributes, 'name' | 'as'>): string {
 	return family.as ?? family.name;
 }
+
+/** Resolves the font family provider. If none is provided, it will infer the correct one and resolve it. */
+export async function resolveFontFamily({
+	family,
+	...resolveProviderOptions
+}: Omit<ResolveProviderOptions, 'provider'> & {
+	family: FontFamily<BuiltInProvider | FontProvider>;
+}): Promise<ResolvedFontFamily> {
+	if (family.provider === LOCAL_PROVIDER_NAME || (!family.provider && 'src' in family)) {
+		return {
+			...family,
+			provider: LOCAL_PROVIDER_NAME,
+		};
+	}
+
+	const provider =
+		family.provider === GOOGLE_PROVIDER_NAME || !family.provider
+			? await import('./providers/google.js').then((mod) => mod.google())
+			: family.provider;
+
+	return {
+		...family,
+		provider: await resolveProvider({
+			...resolveProviderOptions,
+			provider,
+		}),
+	};
+}
+
+function sortObjectByKey<T extends Record<string, any>>(unordered: T): T {
+	const ordered = Object.keys(unordered)
+		.sort()
+		.reduce((obj, key) => {
+			// @ts-expect-error
+			obj[key] = unordered[key];
+			return obj;
+		}, {} as T);
+	return ordered;
+}
+
+/**
+ * Extracts providers from families so they can be consumed by unifont.
+ * It deduplicates them based on their config and provider name, to eg.
```

Comment:
```suggestion
 * It deduplicates them based on their config and provider name, to i.e.
```

This feels more like an `i.e.`, not a `e.g.`: https://www.merriam-webster.com/grammar/ie-vs-eg-abbreviation-meaning-usage-difference

Also, please provide more information about the heuristic used to de-duplicate the providers. This is important information, because "dedupe" doesn't mean anything without proper context (we don't use the heuristic of the package managers).